### ISSUE TITLE
Add Support for Google Tag Manager

### ISSUE THREAD

**Issue by dmolsen** [wvuweb/cleanslate-cms#219]
I'm unsure how we'd like to handle adding support for Google Tag Manager. It requires two separate snippets of JS to work, one in the `<head>` and one in `<body>`. That's in addition to the tag manager container ID. Is this something where we can have explicit tags to include the relevant code? Maybe something like:

```
<r:gtm:head />
and
<r:gtm:body />
```

This would leave it up to the theme developers to put it where it makes sense. We'd have to add it to the admin UI as well in case anyone with the Super Theme wants to opt in to Tag Manager. I would only do that for groups that are actively advertising across platforms or paying attention to analytics.

Oh, and the tags will need to check if that var is set before writing out the code.

For `<r:gtm:head />`:

``` javascript
<!-- Google Tag Manager -->
<script>(function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
})(window,document,'script','dataLayer','{$container_id}');</script>
<!-- End Google Tag Manager -->
```

For `<r:gtm:body />`:

``` javascript
<!-- Google Tag Manager (noscript) -->
<noscript><iframe src="https://www.googletagmanager.com/ns.html?id={$container_id}"
height="0" width="0" style="display:none;visibility:hidden"></iframe></noscript>
<!-- End Google Tag Manager (noscript) -->
```

**Comment by nreckart** [comment]
@dmolsen Is this still relevant?

**Comment by dmolsen** [comment]
Yes and no.

I don't think this needs to be explicit as a tag anymore. We've only added a few sites to GTM in _checks ticket date_ 5+ years. The only thing I've noticed from those few people we have manually done it for, are complaints about the GA warning banner. Since a site is live and doesn't have GA added it gets a yellow banner in the CleanSlate Admin. A checkbox on the site possibly? Just to kill the warning.
